### ISSUE TITLE
Syncfiles fails on diskless nodes as a ".statelite" folder is always made at genimage time

### ISSUE THREAD

**Issue by arjones85** [xcat2/xcat-core#3244]
On booting up a diskless node, I discovered that the genimage process is always creating a ".statelite" folder in the root of the image. As a result, syncfiles sees this folder and fails to run as it believes the node is a statelite node.

Please fix. Thanks!

**Comment by immarvin** [comment]
hi @bybai , would you please take a look at this?

**Comment by bybai** [comment]
Hi @arjones85 , thanks you found this issue, will fix this soon.

**Comment by bybai** [comment]
Hi @arjones85,  Could you comment these code in `/install/postscripts/syncfiles`? that can work around this issue.  
```
if [ -d /.statelite ]; then
   echo "Statelite does not support syncfiles, nothing to do..."
   logger -t xcat -p local4.info "Statelite does not support syncfiles, nothing to do..."
   exit 0
fi
```

**Comment by bybai** [comment]
I looked into `genimage`, after `genimage`, there is `rootimg/.statelite` dir, since genimage works for both statelite and stateless, so I will enhance `packimage` to exclude this `rootimg/.statelite` dir.

**Comment by bybai** [comment]
Hi @arjones85, I tried to reproduce this issue in my environment, but I cannot reproduce it. I want to confirm the following:
1) Could you give me how to reproduce this issue detail steps?
2) What is your xcat version and OS version?

After `genimage` command, there is `rootimg/.statelite` dir; Then I execute `packaimge <osimage>` command, `packimage` does not pack `rootimg/.statelite` dir into rootimg.cpio.gz, after the diskless compute node is booted, there is no /.statelite dir in compute node. The related code in packimage.pm is:
```
    # the files specified for statelite should be excluded
    my @excludeStatelite = ("./etc/init.d/statelite", "./etc/rc.sysinit.backup", "./.statelite*", "./.default*", "./.statebackup*");
    foreach my $entry (@excludeStatelite) {
        $excludestr .= "'!' -path '" . $entry . "' -a ";
    }
```

**Comment by arjones85** [comment]
Hello,

I am out this week, however commenting out the if statement for checking if .statelite exits is what I did to correct the issue. The packimage *is* including the statelite folder in the gzip for my version of xcat.

I can get you the exact version next week when I am back in town. Thanks!

⁣

**Comment by bybai** [comment]
Hi @arjones85 Could you provide the version of xCAT ?  Could you check if there is `./.statelite*` is in packimage.pm like I described in above comments? Thanks

**Comment by arjones85** [comment]
Hello,

My problematic xCAT version is 2.13.3.

> After genimage command, there is rootimg/.statelite dir;

This is correct and consistent with what I see.

> Then I execute packaimge <osimage> command, packimage does not pack rootimg/.statelite dir into rootimg.cpio.gz, after the diskless compute node is booted, there is no /.statelite dir in compute node

This does not happen for me, the .statelite directory is still there in the archive. My packimage.pm dpes have the following in it:

    # the files specified for statelite should be excluded
    my @excludeStatelite = ("./etc/init.d/statelite", "./etc/rc.sysinit.backup", "./.statelite*", "./.default*", "./.statebackup*");
    foreach my $entry (@excludeStatelite) {
        $excludestr .= "'!' -path '" . $entry . "' -a ";
    }


Example I just did. Note that the .statelite directory is extracted from the created cpio archive. This is consistent with what my nodes have in their root directory on boot.

`[root@evxcat rootimg]# pwd
/install/netboot/centos6.6/x86_64/v6.6.0-dl-login-rdhpc/rootimg

[root@evxcat rootimg]# ls -al
total 184
dr-xr-xr-x  27 root root  4096 Jun 20 14:09 .
drwxr-xr-x   3 root root  4096 Jun 20 14:09 ..
dr-xr-xr-x   2 root root 12288 Jun  9 10:02 bin
dr-xr-xr-x   3 root root  4096 Jun  9 10:01 boot
drwxr-xr-x   2 root root  4096 Jun  9 09:49 dev
drwxr-xr-x 114 root root 20480 Jun  9 10:05 etc
drwxr-xr-x   2 root root  4096 Sep 23  2011 home
dr-xr-xr-x  10 root root  4096 Jun  9 09:39 lib
dr-xr-xr-x   9 root root 20480 Jun  9 09:48 lib64
drwxr-xr-x   2 root root  4096 Jun  9 10:02 lustre4
drwxr-xr-x   2 root root  4096 Sep 23  2011 media
drwxr-xr-x   2 root root  4096 Oct 15  2014 misc
drwxr-xr-x   2 root root  4096 Sep 23  2011 mnt
drwxr-xr-x   2 root root  4096 Oct 15  2014 net
drwxr-xr-x  13 root root  4096 Jun  9 10:05 opt
dr-xr-xr-x   2 root root  4096 Sep 23  2011 proc
dr-xr-x---   2 root root  4096 Jun  9 09:49 root
dr-xr-xr-x   2 root root 24576 Jun  9 09:52 sbin
drwxr-xr-x   2 root root  4096 Sep 23  2011 selinux
drwxr-xr-x   4 root root  4096 Jun  9 10:02 .sllocal
drwxr-xr-x   2 root root  4096 Sep 23  2011 srv
**drwxr-xr-x   2 root root  4096 Jun  9 10:02 .statelite**
drwxr-xr-x   2 root root  4096 Sep 23  2011 sys
drwxrwxrwt   5 root root  4096 Jun  9 10:04 tmp
drwxr-xr-x  16 root root  4096 Jun  9 10:05 usr
drwxr-xr-x  23 root root  4096 Jun  9 10:02 var
lrwxrwxrwx   1 root root    18 Jun  9 10:02 workdev -> /data/gpfs/workdev
drwxr-xr-x   8 root root 16384 Jun 20 14:09 xcatpost
[root@evxcat rootimg]# cd ..
[root@evxcat v6.6.0-dl-login-rdhpc]# 

[root@evxcat v6.6.0-dl-login-rdhpc]# ll
total 2070020
-rw-------  1 root root   24939431 Jun  9 10:03 initrd-stateless.gz
-rw-------  1 root root   22297171 Jun  9 10:04 initrd-statelite.gz
-rw-r--r--  1 root root    4152336 Jun  9 09:49 kernel
dr-xr-xr-x 27 root root       4096 Jun  9 10:05 rootimg
-rw-r--r--  1 root root 2059970670 Jun  9 10:13 rootimg.cpio.gz
[root@evxcat v6.6.0-dl-login-rdhpc]# rm -f rootimg.cpio.gz

[root@evxcat v6.6.0-dl-login-rdhpc]# ll
total 50400
-rw-------  1 root root 24939431 Jun  9 10:03 initrd-stateless.gz
-rw-------  1 root root 22297171 Jun  9 10:04 initrd-statelite.gz
-rw-r--r--  1 root root  4152336 Jun  9 09:49 kernel
dr-xr-xr-x 27 root root     4096 Jun  9 10:05 rootimg

[root@evxcat v6.6.0-dl-login-rdhpc]# packimage v6.6.0-dl-login-rdhpc
Packing contents of /install/netboot/centos6.6/x86_64/v6.6.0-dl-login-rdhpc/rootimg
archive method:cpio
compress method:gzip

[root@evxcat v6.6.0-dl-login-rdhpc]# ll
total 2070016
-rw-------  1 root root   24939431 Jun  9 10:03 initrd-stateless.gz
-rw-------  1 root root   22297171 Jun  9 10:04 initrd-statelite.gz
-rw-r--r--  1 root root    4152336 Jun  9 09:49 kernel
dr-xr-xr-x 27 root root       4096 Jun 20 14:09 rootimg
-rw-r--r--  1 root root 2059965919 Jun 20 14:16 rootimg.cpio.gz


[root@evxcat v6.6.0-dl-login-rdhpc]# cp rootimg.cpio.gz /tmp/
[root@evxcat v6.6.0-dl-login-rdhpc]# cd /tmp
[root@evxcat tmp]# gzip -d rootimg.cpio.gz 


**NOTE - No .statelite directory below**
[root@evxcat tmp]# ls -al
total 4787516
drwxrwxrwt  4 root     root           4096 Jun 20 14:19 .
drwxr-xr-x 35 root     root           4096 Apr 18 14:50 ..
drwxrwxrwt  2 root     root           4096 Jul 27  2015 .ICE-unix
srw-------  1 root     root              0 Aug 19  2016 .java_pid23520
-rw-r--r--  1 root     root     4902391296 Jun 20 14:17 rootimg.cpio
srwxrwxrwx  1 postgres postgres          0 Jun 20 13:29 .s.PGSQL.5432
-rw-------  1 postgres postgres         26 Jun 20 13:29 .s.PGSQL.5432.lock
drwxrwxrwt  2 root     root           4096 Apr 18 14:50 .X11-unix

[root@evxcat tmp]# cpio -id < rootimg.cpio 

**NOTE: There is now a .statelite directory below**
[root@evxcat tmp]# ls -al
total 4787648
drwxrwxrwt  29 root     root           4096 Jun 20 14:20 .
drwxr-xr-x  35 root     root           4096 Apr 18 14:50 ..
dr-xr-xr-x   2 root     root           4096 Jun 20 14:20 bin
dr-xr-xr-x   3 root     root           4096 Jun 20 14:20 boot
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 dev
drwxr-xr-x 114 root     root          12288 Jun 20 14:20 etc
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 home
drwxrwxrwt   2 root     root           4096 Jul 27  2015 .ICE-unix
srw-------   1 root     root              0 Aug 19  2016 .java_pid23520
dr-xr-xr-x  10 root     root           4096 Jun 20 14:20 lib
dr-xr-xr-x   9 root     root          12288 Jun 20 14:20 lib64
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 lustre4
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 media
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 misc
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 mnt
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 net
drwxr-xr-x  13 root     root           4096 Jun 20 14:20 opt
dr-xr-xr-x   2 root     root           4096 Jun 20 14:20 proc
dr-xr-x---   2 root     root           4096 Jun 20 14:20 root
-rw-r--r--   1 root     root     4902391296 Jun 20 14:17 rootimg.cpio
dr-xr-xr-x   2 root     root          12288 Jun 20 14:20 sbin
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 selinux
drwxr-xr-x   4 root     root           4096 Jun 20 14:20 .sllocal
srwxrwxrwx   1 postgres postgres          0 Jun 20 13:29 .s.PGSQL.5432
-rw-------   1 postgres postgres         26 Jun 20 13:29 .s.PGSQL.5432.lock
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 srv
**drwxr-xr-x   2 root     root           4096 Jun 20 14:20 .statelite**
drwxr-xr-x   2 root     root           4096 Jun 20 14:20 sys
drwxrwxrwt   5 root     root           4096 Jun 20 14:20 tmp
drwxr-xr-x  16 root     root           4096 Jun 20 14:20 usr
drwxr-xr-x  23 root     root           4096 Jun 20 14:20 var
lrwxrwxrwx   1 root     root             18 Jun 20 14:20 workdev -> /data/gpfs/workdev
drwxrwxrwt   2 root     root           4096 Apr 18 14:50 .X11-unix
drwxr-xr-x   8 root     root          12288 Jun 20 14:20 xcatpost



`

**Comment by bybai** [comment]
Hi @arjones85 , thanks your details comments. I found there was no `./.default` and `./.statebackup` under rootimg. In packimage.pm, directories `./.statelite`, `./.default` and `./.statebackup` are all in exclude list, why only left `./.statelite` directory? It is curious to me, I will look into it later.

**Comment by zet809** [comment]
Move the next release for more investigation.

**Comment by whowutwut** [comment]
@zet809  Let's start to allow owners to modify the milestones based on their prioritization of the work.   @bybai If this is something that is not in your scope of work, you can remove the milestone or pick one that is further out but one that we can really have a harder target on.  If the investigation is not done, then we probably should not target the next sprint for completion.

**Comment by zet809** [comment]
OK, I agree.

**Comment by bybai** [comment]
Hi @arjones85 ,  I fixed issue #3912.
Could you check if there is no exlist in your osimage definition?
Actually, we recommend there should be exlist in osimage definition, or else, the rootimg is too large for stateless.
In pull request #3919, I enhanced `packimage`, so it should also fix your issue.
